xdsl/transforms/experimental/stencil_tensorize_z_dimension.py

```diff
@@ -224,16 +224,15 @@ def match_and_rewrite(self, op: ApplyOp, rewriter: PatternRewriter, /):
                         access_op.offset.array.data[-1].data + z_shift,
                     )
 
-            # TODO check if block args need updating
-            for _ in range(len(b.args)):
-                assert isa(arg_type := b.args[0].type, TempType[Attribute])
-                arg = b.insert_arg(stencil_temp_to_tensor(arg_type), len(b.args))
-                b.args[0].replace_by(arg)
-                b.erase_arg(b.args[0], safe_erase=True)
+            body = Block(arg_types=op.operand_types)
+            rewriter.inline_block(
+                op.region.block, InsertPoint.at_start(body), body.args
+            )
```

Comment:
Instead of updating args individually, rebuild from operand types. Simpler, and a bug fix.